fs2loader/src/main/scala/com.snowplowanalytics.snowplow.storage.bigquery.fs2loader/Fs2Loader.scala

```diff
@@ -0,0 +1,240 @@
+import com.snowplowanalytics.snowplow.storage.bigquery.fs2loader.sinks.{Bigquery, PubSub}
+import com.snowplowanalytics.snowplow.storage.bigquery.fs2loader.sinks.PubSub.{WriteBadRow, WriteObservedTypes}
+import com.snowplowanalytics.snowplow.storage.bigquery.loader.LoaderRow
+import com.snowplowanalytics.snowplow.analytics.scalasdk.Data.ShreddedType
+import com.google.cloud.bigquery.BigQuery
+
+import scala.concurrent.duration._
+
+object Fs2Loader {
+
+//      TOPOLOGY OVERVIEW
+//                                            +-----------+
+//                                 bad sink   |   PubSub  |
+//                              +-------------|   (bad)   |
+//                              |             +-----------+
+//                              |
+//  +------------+          +--------+                                       aggregate
+//  |            |  parse   | (bad,  |                  enqueue   +-------+  >> sink     +--------+
+//  | enriched   | -------- | good)  |                  types     |types  |  types       | PubSub |
+//  |            |          |        |               +------------|queue  | -------------| (types)|
+//  +------------+          +--------+               |            +-------+              +--------+
+//                              |                    |
+//                              |                    |
+//                              |             +------------+                              +---------+
+//                              |             |            |              failed inserts  | PubSub  |
+//                              +-------------| good pipe  |             +----------------| (failed |
+//                                            |            |             |                | inserts)|
+//                                            +------|-----+             |                +---------+
+//                                                   |                   |
+//                                                   |             +-----|----+
+//                                                   |             | BigQuery |
+//                                                   +-------------| pipe     |
+//                                                                 +-----|----+
+//                                                                       |
+//                                                                       |                 +--------+
+//                                                                       |                 |BigQuery|
+//                                                                       +-----------------|(events)|
+//                                                                                         +--------+
+
+  private val MaxConcurrency = 10
+  private val GroupByN       = 10
+  private val TimeWindow     = 10.seconds
+
+  def badSink(badRow: BadRow)(env: Environment): IO[Unit] =
```

Comment:
It's not a sink technically. Sink is `Pipe[F, A, Unit]`. This is just a core function of a sink, but sink implies some relation to stream.